### PR TITLE
backported changes to develop

### PR DESCRIPTION
**Pull request by AlmirKadric**
Currently the COJ branch has been heavily refactored and work or bug fixes have been applied both there and on develop. This PR backports changes from Develop and in some instances reverts refactoring done as they introduce bugs or were not expected. The idea is to bring both branches to a unification so that no work is lost.

### CODE REVIEW

File: EditorScript/UnityApplicationState.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	{
+		public delegate void AppStateChanged(bool pauseStatus);
+		public AppStateChanged OnAppStateChanged;
+
+		// What should happen when the application is put into the background
+		void OnApplicationPause(bool pauseStatus)
+		{
+			if (OnAppStateChanged != null)
+			{
+                OnAppStateChanged.Invoke(pauseStatus);
```
Comment (nullorvoid) [inline EditorScript/UnityApplicationState.cs]: spaces!

File: MageClient/Command/Client/CommandJsonrpcClient.cs [hunk 2]
```diff
@@ -17,16 +18,22 @@ private Mage Mage
 
 		private Logger Logger
 		{
-			get { return Mage.Logger("CommandJSONRPCClient"); }
+			get { return Mage.Logger("CommandJsonrpcClient"); }
```
Comment (nullorvoid) [inline MageClient/Command/Client/CommandJsonrpcClient.cs]: same r

File: MageClient/Message/Client/ShortPolling.cs [hunk 3]
```diff
@@ -109,7 +107,7 @@ private void RequestLoop()
 			// Send poll request and wait for a response
 			string endpoint = getEndpoint();
 			Logger.Debug("Sending request: " + endpoint);
-			HttpRequest.Get(endpoint, getHeaders(), Mage.Cookies, (requestError, responseString) => {
+            HttpRequest.Get(endpoint, getHeaders(), Mage.Cookies, (requestError, responseString) => {
```
Comment (nullorvoid) [inline MageClient/Message/Client/ShortPolling.cs]: spaces

**Comment by kefniark** [comment]
this one looks kinda OK

**Comment by AlmirKadric** [comment]
added a description @kefniark ;)

**Comment by nullorvoid** [comment]
LGTM

**Pull request merged**